itests/kit/node_unmanaged.go

```diff
@@ -24,3 +60,730 @@ type TestUnmanagedMiner struct {
+
+	tm.sealTickets[sectorNumber] = sealTickets
+	tm.sealedCids[sectorNumber] = sealedCid
+	tm.unsealedCids[sectorNumber] = unsealedCid
+}
+
+func (tm *TestUnmanagedMiner) proveCommitWaitSeed(ctx context.Context, sectorNumber abi.SectorNumber) abi.InteractiveSealRandomness {
+	req := require.New(tm.t)
+	head, err := tm.FullNode.ChainHead(ctx)
+	req.NoError(err)
+
+	tm.t.Logf("Miner %s: Fetching pre-commit info for sector %d...", tm.ActorAddr, sectorNumber)
+	preCommitInfo, err := tm.FullNode.StateSectorPreCommitInfo(ctx, tm.ActorAddr, sectorNumber, head.Key())
+	req.NoError(err)
+	seedRandomnessHeight := preCommitInfo.PreCommitEpoch + policy.GetPreCommitChallengeDelay()
+
+	tm.t.Logf("Miner %s: Waiting %d epochs for seed randomness at epoch %d (current epoch %d) for sector %d...", tm.ActorAddr, seedRandomnessHeight-head.Height(), seedRandomnessHeight, head.Height(), sectorNumber)
+	tm.FullNode.WaitTillChain(ctx, HeightAtLeast(seedRandomnessHeight+5))
+
+	minerAddrBytes := new(bytes.Buffer)
+	req.NoError(tm.ActorAddr.MarshalCBOR(minerAddrBytes))
+
+	head, err = tm.FullNode.ChainHead(ctx)
+	req.NoError(err)
+
+	tm.t.Logf("Miner %s: Fetching seed randomness for sector %d...", tm.ActorAddr, sectorNumber)
+	rand, err := tm.FullNode.StateGetRandomnessFromBeacon(ctx, crypto.DomainSeparationTag_InteractiveSealChallengeSeed, seedRandomnessHeight, minerAddrBytes.Bytes(), head.Key())
+	req.NoError(err)
+	seedRandomness := abi.InteractiveSealRandomness(rand)
+
+	tm.t.Logf("Miner %s: Obtained seed randomness for sector %d: %x", tm.ActorAddr, sectorNumber, seedRandomness)
+	return seedRandomness
+}
+
+func (tm *TestUnmanagedMiner) generateProveCommit(
+	ctx context.Context,
+	sectorNumber abi.SectorNumber,
+	proofType abi.RegisteredSealProof,
+	seedRandomness abi.InteractiveSealRandomness,
+	pieces []abi.PieceInfo,
+) []byte {
+	tm.t.Logf("Miner %s: Generating proof type %d Sector Proof for sector %d...", tm.ActorAddr, proofType, sectorNumber)
+	req := require.New(tm.t)
+
+	actorIdNum, err := address.IDFromAddress(tm.ActorAddr)
+	req.NoError(err)
+	actorId := abi.ActorID(actorIdNum)
+
+	tm.t.Logf("Miner %s: Running proof type %d SealCommitPhase1 for sector %d...", tm.ActorAddr, proofType, sectorNumber)
+
+	scp1, err := ffi.SealCommitPhase1(
+		proofType,
+		tm.sealedCids[sectorNumber],
+		tm.unsealedCids[sectorNumber],
+		tm.cacheDirPaths[sectorNumber],
+		tm.sealedSectorPaths[sectorNumber],
+		sectorNumber,
+		actorId,
+		tm.sealTickets[sectorNumber],
+		seedRandomness,
+		pieces,
+	)
+	req.NoError(err)
+
+	tm.t.Logf("Miner %s: Running proof type %d SealCommitPhase2 for sector %d...", tm.ActorAddr, proofType, sectorNumber)
+
+	sectorProof, err := ffi.SealCommitPhase2(scp1, sectorNumber, actorId)
+	req.NoError(err)
+
+	tm.t.Logf("Miner %s: Got proof type %d sector proof of length %d for sector %d", tm.ActorAddr, proofType, len(sectorProof), sectorNumber)
+
+	return sectorProof
+}
+
+func (tm *TestUnmanagedMiner) submitMessage(
```

Comment:
I think this should be refactored to return an error rather than using `tm.t` because it's used inside the PoST goroutine

Reply:
Done.